numpy/_core/strings.py

```diff
@@ -139,15 +139,40 @@ def multiply(a, i):
            ['d', 'ee', 'fff']], dtype='<U3')
            
     """
-    a_arr = np.asarray(a)
-    i_arr = np.asarray(i)
-    if not issubclass(i_arr.dtype.type, np.integer):
-        raise ValueError("Can only multiply by integers")
-    if a_arr.dtype.kind == "T":
-        return a_arr * i_arr
-    out_size = _get_num_chars(a_arr) * max(int(i_arr.max()), 0)
-    return _vec_string(
-        a_arr, type(a_arr.dtype)(out_size), '__mul__', (i_arr,))
+    a = np.asanyarray(a)
+    i = np.asanyarray(i)
+    a, i = np.broadcast_arrays(a, i)
+    if not np.issubdtype(i.dtype, np.integer):
+        raise TypeError(f"unsupported type {i.dtype} for operand 'i'")
+
+    # (negative number) * (any string dtype) should return an empty string
+    i = np.where(i < 0, 0, i)
+
+    # get length of input elements for computing the output buffer sizes
+    try:
+        a_len = str_len(a)
+    except ValueError:
+        # dtype is StringDType and there are nan strings
+        a_len = np.full(a.shape, 0)
+        non_nan = np.logical_not(np.isnan(a))
+        a_len[non_nan] = str_len(a[non_nan])
```

Comment:
sorry I missed this on the first pass, you can delete this if you make stringdtype just delegate to the ufunc before this check,